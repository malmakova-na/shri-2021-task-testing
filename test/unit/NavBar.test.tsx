import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import events from "@testing-library/user-event";
import { Router } from "react-router";
import { Provider } from "react-redux";
import { createMemoryHistory } from 'history';

import { Application } from "../../src/client/Application";
import { ExampleApi, CartApi } from "../../src/client/api";
import { initStore } from "../../src/client/store";

it("Правильно отображается NavBar", () => {
    const history = createMemoryHistory({
        initialEntries: ['/'],
        initialIndex: 0
        });
        
        const basename = "/hw/store";
        
        const api = new ExampleApi(basename);
        const cart = new CartApi();
        const store = initStore(api, cart);
        
        const application = (
        <Router history={history}>
            <Provider store={store}>
                <Application />
            </Provider>
        </Router>
        );
        const {getByRole, container} = render(application);
        
        expect(screen.getByRole('link', {
            name: /catalog/i
        })).toBeInTheDocument();

        expect(screen.getByRole('link', {
            name: /delivery/i
        })).toBeInTheDocument();

        expect(screen.getByRole('link', {
            name: /contacts/i
        })).toBeInTheDocument();

        expect(screen.getByRole('link', {
            name: /cart/i
        })).toBeInTheDocument();

});