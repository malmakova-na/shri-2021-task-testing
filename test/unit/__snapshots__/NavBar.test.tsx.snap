// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Правильно отображается NavBar 1`] = `
<div
  className="Application"
>
  <nav
    className="navbar navbar-expand-sm navbar-light bg-light"
  >
    <div
      className="container"
    >
      <a
        className="Application-Brand navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Example store
      </a>
      <button
        aria-label="Toggle navigation"
        className="Application-Toggler navbar-toggler"
        onClick={[Function]}
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="Application-Menu collapse navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/catalog"
            onClick={[Function]}
          >
            Catalog
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/delivery"
            onClick={[Function]}
          >
            Delivery
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/contacts"
            onClick={[Function]}
          >
            Contacts
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/cart"
            onClick={[Function]}
          >
            Cart
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container pt-4"
  >
    <div
      className="Home"
    >
      <div
        className="row"
      >
        <div
          className="col bg-secondary text-white py-4 bg-opacity-75"
        >
          <p
            className="display-3"
          >
            Welcome to Example store!
          </p>
          <p
            className="lead"
          >
            Culpa perspiciatis corporis facilis fugit similique
          </p>
          <p
            className="lead"
          >
            Cum aut ut eveniet rem cupiditate natus veritatis quia
          </p>
        </div>
      </div>
      <div
        className="row mb-4"
      >
        <div
          className="col-12 col-md-4 bg-light py-3"
        >
          <h1>
            Quickly
          </h1>
          <p
            className="lead"
          >
            Odio aut assumenda ipsam amet reprehenderit. Perspiciatis qui molestiae qui tempora quisquam
          </p>
        </div>
        <div
          className="col-12 col-md-4 bg-light py-3"
        >
          <h1>
            Qualitatively
          </h1>
          <p
            className="lead"
          >
            Ut nisi distinctio est non voluptatem. Odio aut assumenda ipsam amet reprehenderit
          </p>
        </div>
        <div
          className="col-12 col-md-4 bg-light py-3"
        >
          <h1>
            Inexpensive
          </h1>
          <p
            className="lead"
          >
            Perspiciatis qui molestiae qui tempora quisquam. Ut nisi distinctio est non voluptatem
          </p>
        </div>
      </div>
      <div
        className="row mb-4"
      >
        <div
          className="col-12py-3"
        >
          <p>
            Sed voluptatum quis voluptates laudantium incidunt laudantium. Illo non quos eos vel ipsa. Explicabo itaque est optio neque rerum provident enim qui sed. Corrupti commodi voluptatem vero soluta hic.
          </p>
          <p>
            Modi corporis consectetur aliquid sit cum tenetur enim. Sed voluptatum quis voluptates laudantium incidunt laudantium. Illo non quos eos vel ipsa. Explicabo itaque est optio neque rerum provident enim qui sed. Corrupti commodi voluptatem vero soluta hic.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
